src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -52,40 +63,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } catch (GoogleJsonResponseException e) { // some other problem with requesting comment data
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, COMMENTS_FAILED_ERROR);
     }
-    String comments = String.join(". ", commentsList);
 
     String captions = captionService.getCaptionFromId(videoId);
 
-    if (comments.isEmpty() && captions.isEmpty()) {
-      VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-          .setScore(null)
-          .setScoreAvailable(false)
-          .build();
-      String json = new Gson().toJson(videoAnalysis);
-
+    if (commentsList.isEmpty() && captions.isEmpty()) {
+      String json = createResponseJson(null, false);
```

Comment:
I recommend you go back explicitly building the response here. The reason why it was made to a builder was to make it easy to read what was happening. Now this helper method makes it hard to read. Not obvious to know what null and false are being assigned to without reading the helper method.

I'm guessing you did this because you called this helper method in different branches below. Instead you could have branches to calculate the total average into one variable and after the branches you can build the response once.